File: elderlyCareSystem-master/elderlyCareSystem-master/cm/ElderlyCareSystem/app/src/main/java/com/example/elderlycaresystem/ui/main/MainActivity.java
package com.example.elderlycaresystem.ui.main;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import com.example.elderlycaresystem.data.elderly.ElderlyInfo;
import com.example.elderlycaresystem.R;
import com.example.elderlycaresystem.ui.login.LoginActivity;
import com.example.elderlycaresystem.util.RetroUtils;

import java.util.List;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {
    public static final String INTENT_ID = "INTENT_ID"; // static -> 다른 액티비티에서도 사용 가능
    public static final String INTENT_KEY= "INTENT_KEY";
    private String id;

    private long time= 0;// System Back Button time
    private MainAdapter adapter;

    @Override
        public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // TODO : 1. LoginActivity에서 Key(ID) 받아오기
        Intent intent = getIntent();
        //TODO : Push로 App 실행 후 MainActivity로 접근 시
        if (intent.getStringExtra(INTENT_ID)==null){
            Intent loginIntent = new Intent(this, LoginActivity.class);
            startActivity(loginIntent);
            finish();
        }else{
            id = intent.getStringExtra(INTENT_ID);
        }


        //String cookie = bundle.getString(INTENT_COOKIE);

        Toast.makeText(MainActivity.this, "User's ID : " + id, Toast.LENGTH_SHORT).show();

        // Adapter를 RecyclerView 객체에 설정하고, Adapter 안에 Elderly 객체들을 넣기!
        // recyclerView 객체 할당
        RecyclerView recyclerView = findViewById(R.id.recyclerView);

        // RecyclerView에 Layout Manager 설정하기
        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
        recyclerView.setLayoutManager(layoutManager);
        adapter = new MainAdapter();

        recyclerView.setAdapter(adapter);
        getElderlyList();

    }

    // TODO : LoopBack version. ( Test version : getTestList() -> getElderlyList(id) )
    private void getElderlyList() {
        RetroUtils.getElderlyService(getApplicationContext()).getElderlyList(id).enqueue(new Callback<List<ElderlyInfo>>() {
            @Override
            public void onResponse(Call<List<ElderlyInfo>> call, Response<List<ElderlyInfo>> response) {
                if (response.isSuccessful() && response.body() != null) {
                    Toast.makeText(MainActivity.this, "수신 성공!", Toast.LENGTH_SHORT).show();
                    adapter.addItems(response.body());
                    adapter.notifyDataSetChanged();// 어댑터로 화면 내 데이터 새로고침
                } else {
                    int statusCode = response.code();
                    Log.d("Main_getList_Empty", "Empty Body.");
                    Toast.makeText(MainActivity.this, "수신 실패! " + statusCode, Toast.LENGTH_SHORT).show();
                }
            }
            @Override
            public void onFailure(Call<List<ElderlyInfo>> call, Throwable t) {
                Log.d("Main_getList_Failure", "Get Failed.");
                Toast.makeText(MainActivity.this, "접속 실패! " + t.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        logoutService();
    }

    // TODO : 액티비티로 돌아올 때마다 데이터 갱신
    @Override
    protected void onResume() {
        super.onResume();
        getElderlyList();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    public void onBackPressed() {
        if (System.currentTimeMillis() - time >= 2000) {
            time = System.currentTimeMillis();
            Toast.makeText(MainActivity.this,"뒤로 가시려면 한번 더 눌러주세요",Toast.LENGTH_LONG).show();
        } else if (System.currentTimeMillis() - time < 2000) {
            finish();
        }
    }

    private void logoutService() {
        RetroUtils.getElderlyService(getApplicationContext()).logout().enqueue(new Callback<ResponseBody>() {
            // 연결 성공시
            @Override
            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
                if(response.isSuccessful() && response.body() != null) {
                    Log.d("Main_logout_Success", "Get Success.");
                }else {
                    int statusCode  = response.code();
                    Log.d("Main_logout_Success", "Empty Body.");
                    Toast.makeText(MainActivity.this, "Error code : " + statusCode, Toast.LENGTH_SHORT).show();
                }
            }
            @Override
            public void onFailure(Call<ResponseBody> call, Throwable t) {
                Log.d("Main_logout_Failure", "Get Failed.");
                Toast.makeText(MainActivity.this, "Error code : " + t.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);

    }
}

File: cm/BIT/app/src/main/java/com/example/bit/MainActivity.kt
package com.example.bit

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.example.bit.Utils.RetrofitUtils
import kotlinx.android.synthetic.main.activity_main.*
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    // myAdapter -> MainActivity의 recycler View에 아이템을 넣어 띄울 MainAdapter 클래스
    var myAdapter = MainAdapter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //TODO : 1. 전체 데이터 수신(ticker/ALL_KRW)
        /** 전체 버튼 클릭 리스너 등록 **/
        showAllButton.setOnClickListener {
            showAllTickerData()
        }
    }

    /**
     * showAllData()
     * @param none
     * @return none
     *  - 전체(110개) Ticker(현재 시세) 데이터 수신
     *  - myAdapter에 110개 데이터를 담아 recycler view에 표시
     */
    private fun showAllTickerData(){
        // RetrofitUtils.getBitService(applicationContext) -> 톹신에 사용될 client 생성 및 BitService return
        // BTCTicker().BTCTicker() -> BTC 종목에 대한 Ticker 데이터를 받아온다. => @GET, URI : .../ticker/BTC_KRW.
        // enqueue( ... ) -> 해당 URL로 요청을 보낸다
        // object : retrofit2.Callback<ResponseBody> { ... } -> 전달 받은 응답(CallBack)에 대한 처리
        RetrofitUtils.getBitService(applicationContext).BTCTicker()?.enqueue(object : retrofit2.Callback<ResponseBody> {
            // 요청이 성공했을 경우(서버에 요청이 전달 된 상태)
            override fun onResponse(call: Call<ResponseBody?>?,response: Response<ResponseBody?>) {
                // 정상 Callback을 받은 경우
                if (response.isSuccessful() && response.body() != null) {
                    Log.d("Response Data", response.body()!!.string())
                    // TODO : 1.1 응답받은 전체 데이터를 MainAdapter에 담아 화면에 보여주기
                    //  + MainAdapter 수정 필요 사항
                    //   1. MainAdapter에서 쓰일 items( MutableList<TickerData> )의 데이터 타입 수정
                    //   2. addItems()의 param(데이터) 수정
                    //myAdapter.addItems(/** RecyclerView에 표현할 MutableList<MyData> 넣기!! **/)

                    // Adapter로 화면 내 데이터 새로고침
                    myAdapter.notifyDataSetChanged()


                }
                // 정상 Callback을 받지 못한 경우( ex. 404 error )
                else {
                    Log.d("MainActivity", "Not Successful or Empty Response")
                }
            }
            // 요청이 실패했을 경우(서버에 요청이 전달되지 못한 상태)
            override fun onFailure(
                call: Call<ResponseBody?>?, t: Throwable ) {
                Log.d("MainActivity", "Connect_Error")
                t.printStackTrace()
            }
        })
    }


}

